avsomr/ui/recognitionproccessdialog.cpp

```diff
@@ -0,0 +1,128 @@
+#include "recognitionproccessdialog.h"
+
+#include <QPushButton>
+
+using namespace Ms::Avs;
+
+RecognitionProccessDialog::RecognitionProccessDialog()
+      : QProgressDialog()
+      {
+      setWindowFlags(windowFlags() & ~Qt::WindowCloseButtonHint);
+
+      setFixedSize(260, 120);
+      setAutoReset(false);
+
+      _closeBtn = new QPushButton(QObject::tr("Close"));
+      setCancelButton(_closeBtn);
+
+      setRange(0, 100);
+      setValue(0);
+
+      _updater.setInterval(1000);
+      _updater.setSingleShot(false);
+
+      QObject::connect(&_updater, &QTimer::timeout, [this]() {
+            update();
+            });
+      }
+
+//---------------------------------------------------------
+//   formatStep
+//---------------------------------------------------------
+
+QString RecognitionProccessDialog::formatStep(const IAvsOmrRecognizer::Step& step) const
+      {
+      switch (step.type) {
+            case IAvsOmrRecognizer::Step::Undefined:
+                  return QString();
+            case IAvsOmrRecognizer::Step::PrepareStart:
+                  return QObject::tr("Preparing...");
+            case IAvsOmrRecognizer::Step::PrepareFinish:
+                  if (step.success)
+                        return QObject::tr("Success prepare");
+
+                  return QObject::tr("Failed prepare");
+            case IAvsOmrRecognizer::Step::ProcessingStart:
+                  return QObject::tr("Procesing...");
+            case IAvsOmrRecognizer::Step::ProcessingFinish:
+                  if (step.success)
+                        return QObject::tr("Success procesing");
+
+                  return QObject::tr("Failed procesing");
+            case IAvsOmrRecognizer::Step::LoadStart:
+                  return QObject::tr("Loading...");
+            case IAvsOmrRecognizer::Step::LoadFinish:
+                  if (step.success)
+                        return QObject::tr("Success loading");
+
+                  return QObject::tr("Failed loading");
+            }
+      return QString();
+      }
+
+//---------------------------------------------------------
+//   setType
+//---------------------------------------------------------
+
+void RecognitionProccessDialog::setType(const QString& type)
+      {
+      _type = type;
```

Comment:
You may want to put the one-line functions in header file...

Reply:
If possible, I would prefer not to. I usually don’t put the implementation of methods, even simple ones, in .h file. Except for implementations of simple methods child strucs.
If the implementation is in .h file, then it litter the .h file (and it must be clean to easily see the class interface), and this adds complexity for debugging.

Reply:
I really don't think so. Putting all those getters and setters in .cpp file causes very serious clutter to me. It is not necessary. Just put `setType(const QString& type) { _type = type; }` in .h file, how is this complex or litter?

Reply:
> how is this complex or litter?   
  
a few examples that I constantly encounter in my work:
* It is difficult to find a method implementation. The IDE usually has a list of methods in the .cpp file (or by search), and when you do not find the expected method in it, it will confuse you.
* This makes the code inconsistent, part of the implementations in the .cpp, part in the .h, and you do not know where, in fact, you can only guess. When in .h only the declaration, and in the .cpp only the implementation, then everything is clear.
* It's hard to read a .h file, it gets complicated and littered ([ove.h ](https://github.com/musescore/MuseScore/blob/master/mscore/ove.h)  nice to read)  
* It is difficult to add comments to methods in .h, you cannot write a short comment on the same line (like this `QByteArray readMxl (); // compressed MuzicXml`) 
* It is difficult to add verification of input arguments, boundary conditions, state - this requires more than one line, and you have to either not do this, or even litter .h file, or write unreadable code.
* Difficult to debug:
* It is difficult to add logging to see what values and when are assigned
* It is difficult to add debugging code - checking for a specific value, checking for changes in the value.  With this, I tormented myself a lot and for a long time when I found out why the elements are not placed where they should, who, when and which sets the positions for the elements.

Reply:
I know what you're talking about, but what I'm saying is that you only put the getters and setters into the header files. Because just by guessing you know what these public ports are doing. You won't have to search for the definitions too, because you know what they're doing.

Reply:
> Putting all getters and setters in .h file makes the code more consistent too. I know I mention consistency a lot of times, but it's truly one serious reason to consider.

Some decisions were made a long time ago, in other conditions - with other developer experience, with other code base size, other evolution of the software industry.
Everything has changed a long time ago, and the software industry has evolved rapidly (including C ++ over the past 10 years), and MU has grown rapidly by all criteria.
Strict adherence to the old agreement leads to stagnation. And also, we get code that violates all the principles of development. This is not criticism, this is my expert opinion, as a professional and experienced developer.
You can see for yourself. Read about [SOLID](https://en.wikipedia.org/wiki/SOLID) and apply the principles described to MU. (I'm going to write an article about this)

Reply:
OK, so we're going to change this aspect of MuseScore coding styles too? You can certainly add this proposal to the new page (you must know that one), but until the change is approved by the majority of developers, I think it's not wise to simply violate the conventional style.

Reply:
It seems I did not find such a rule, so I’m not breaking anything :)
Yes, I will make a proposal to explicitly review this aspect and fix it in the rules.

Reply:
Yes there's no such _rule_, but it's a _convention_, that's the word I've been using.

Reply:
> In fact, I've already done a clean-up PR #5680 to reorganize these two files and importove.cpp.  

It may be better to fix the place of the implementations in the .h file?, after you now know what problems it causes.

Reply:
"Legalize it!" reminds me of my youth ;-)

Reply:
> It may be better to fix the place of the implementations in the .h file?

Don't quite get you.

Reply:
@Jojo-Schmitz :))

Reply:
> It may be better to fix the place of the implementations in the .h file?, after you now know what problems it causes.

I won't before it's approved by the majority of developers, and I don't see "what problems it causes" not because you didn't explain, but because I've seen no real-world examples of the problems. All've been going pretty well for me.